### PR TITLE
Prevent template run for sync builds, Update SetTestPipelineVersion.ps1

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Prevent template pipeline run triggered by eng/common changes.
Use BuildID as preview version in template runs.
Update SetTestPipelineVersion.ps1

### CODE REVIEW

File: eng/scripts/SetTestPipelineVersion.ps1 [hunk 1]
```diff
@@ -1,28 +1,18 @@
...
 
-$semVarsSorted = [AzureEngSemanticVersion]::SortVersionStrings($semVars)
-LogDebug "Last Published Version $($semVarsSorted[0])"
+. "${PSScriptRoot}\..\common\scripts\common.ps1"
 
-$newVersion = [AzureEngSemanticVersion]::ParseVersionString($semVarsSorted[0])
-$newVersion.IncrementAndSetToPrerelease()
-LogDebug "Version to publish [ $($newVersion.ToString()) ]"
+$changeLogFile = "${PSScriptRoot}\..\..\sdk\template\Azure.Template\CHANGELOG.md"
+$newVersion = "1.0.3-beta.$PreviewVersionNumber"
```
Comment (weshaggard) [inline eng/scripts/SetTestPipelineVersion.ps1:13]: Why are we hardcoding `1.0.3`?

Comment (weshaggard) [reply eng/scripts/SetTestPipelineVersion.ps1:13]: We should be able to use your package parsing methods to get the version and the ChangeLog path.

File: sdk/template/ci.yml [hunk 2]
```diff
@@ -2,24 +2,25 @@
...
+      - feature/*
+      - hotfix/*
+      - release/*
   paths:
     include:
-    - sdk/template/
-    - eng/common/
+      - sdk/template/
+    exclude:
+      - eng/common/
```
Comment (mikeharder) [inline sdk/template/ci.yml:23]: Do you need to exclude `eng/common`?  I think it would be sufficient to just not include it.

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: eng/scripts/SetTestPipelineVersion.ps1 [hunk 3]
```diff
@@ -1,28 +1,18 @@
...
 &"${PSScriptRoot}/Update-PkgVersion.ps1" -ServiceDirectory "template" `
--PackageName 'Azure.Template' -PackageDirName "Azure.Template" -NewVersionString $newVersion.ToString() `
+-PackageName 'Azure.Template' -PackageDirName "Azure.Template" -NewVersionString $newVersion `
 -ReleaseDate (Get-Date -f "yyyy-MM-dd")
-Set-Content -Path $changeLogFile -Value @"
-# Release History
-## $($newVersion.ToString()) ($(Get-Date -f "yyyy-MM-dd"))
-- Test Release Pipeline
-"@
+Set-TestChangeLog -TestVersion $newVersion -changeLogFile $changeLogFile -ReleaseEntry "Test Release Pipeline"
```
[the remark this answers is not in the document]
Comment (chidozieononiwu) [reply eng/scripts/SetTestPipelineVersion.ps1:18]: Am working on another PR.

**Pull request closed without merging**